js/views/network_status_view.js

```diff
@@ -55,20 +58,34 @@
         case WebSocket.CONNECTING:
           message = i18n('connecting');
           this.setSocketReconnectInterval(null);
+          window.clearTimeout(this.connectedTimer);
+          this.connectedTimer = null;
           break;
         case WebSocket.OPEN:
           this.setSocketReconnectInterval(null);
+          window.clearTimeout(this.connectedTimer);
+          this.connectedTimer = null;
           break;
         case WebSocket.CLOSED:
-          message = i18n('disconnected');
-          instructions = i18n('checkNetworkConnection');
-          hasInterruption = true;
+          if (!this.connectedTimer) {
+            // Mark offline if disconnected for 30 seconds
+            this.connectedTimer = window.setTimeout(() => {
+              message = i18n('disconnected');
+              instructions = i18n('checkNetworkConnection');
+              hasInterruption = true;
+            }, DISCONNECTED_DELAY);
+          }
```

Comment:
since we don't really use CLOSING, we can fall through:
```js
case WebSocket.CLOSED:
    // Intentional fall-through
case WebSocket.CLOSING:
default:
    // do your stuff
```
(I like leaving a comment for intentional fall throughs)